### DISCUSSION TITLE
Questions for creating language keyboard layouts

### DISCUSSION THREAD

**Discussion by klaurence** [florisboard/florisboard#2453] [General]
I might (but not for sure) be able to code in some keyboard layouts for other languages. I have some questions to ask beforehand:

- What is the copyright situation on copying or reverse-engineering keyboard layouts from other keyboards? (Is copying/reverse-engineering a layout from e.g. Gboard okay if it has one?)
- What do you use as sources for keyboard layouts? (Sources like websites, papers, etc.)
- What keyboard layout do you implement for a language if there are competing keyboard layouts?

**Comment by patrickgold** [comment]
- Probably depends on the country you reside in as you are the author, however if the copyright question is questionable it is best if you wait for the new layout implementation and the addons store, this way any potential copyright issues only affect the extension and not the whole project
- In the old layout common layouts are mostly taken from other keyboards, in the new implementation we will officially base ourself on the Unicode Keyboard v3 standard for the core layouts, and other layouts will reside in the addons store, which could have any source for the layouts.
- Both and let the user decide?

**Reply by klaurence** [reply]
That makes sense. Thank you.
